src/ir_passes/await.ml

```diff
@@ -127,9 +127,14 @@ and t_exp' context exp =
   | DefineE (id, mut ,exp1) ->
     DefineE (id, mut, t_exp context exp1)
   | FuncE (x, (T.Local as s1), c, typbinds, pat, typs,
-      ({ it = AsyncE _; _} as body)) ->
+      ({ it = AsyncE _; _} as async)) ->
     FuncE (x, s1, c, typbinds, pat, typs,
-      t_async context body)
+      t_async context async)
+  | FuncE (x, (T.Local as s1), c, typbinds, pat, typs,
+      ({it = BlockE (ds, ({ it = AsyncE _; _} as async)); _} as wrapper))->
```

Comment:
Is this doing the right thing even if the user writes a source function that matches that shape (a local function with a block and an `async` block in tail position)?

Reply:
Yeah, I worried about that, but I don't think they can by typing, right?

Reply:
It would actually be nice if desugaring didn't push the pattern match into the body but somehow kept the pattern as part of the function ast, but that's not the current setup.

Reply:
Yeah, feels bad to have these non-compositional invariants around, but for now, LGTM

Reply:
Does this also cover `func () : async () = async { .. }`-style definitions?

Reply:
I'll add a test

Reply:
Done

Reply:
> Yeah, feels bad to have these non-compositional invariants around, but for now, LGTM

Agreed. But if you think this one's bad, you should see the existing case for oneways below.